File: frontend/src/Models/Invoice.ts
class Invoice {
	invoiceNumber: string = "";
	description: string = "";
	issueDate: string = "";
	dueDate?: string | null = null;
	company?: string | null = null;
	customerId?: string | null = null;

	status: number = 2;
	operation: number = 2;
	payment: number = 2;
	currencyType: number = 1;

	products: {
		productId: string;
		pricing: {
			quantity: number;
			unitPrice: number;
			discountRate: number;
			taxRate: number;
			totalPrice: number;
		};
	}[] = [
		{
			productId: "",
			pricing: {
				quantity: 1,
				unitPrice: 0,
				discountRate: 0,
				taxRate: 0,
				totalPrice: 0
			}
		}
	];
}

const InvoiceBodyModel = new Invoice();
export { InvoiceBodyModel };

File: backend/srcs/core/Domain/Entities/CompanyEntities/Invoice.cs
using System.Text.Json.Serialization;
using Domain.Abstractions;
using Domain.Enums;

namespace Domain.Entities.CompanyEntities;

public sealed class Invoice : BaseEntity {
	public Dictionary<string, Guid> Processor     { get; set; }
	public string                   InvoiceNumber { get; set; } = string.Empty;
	public string                   Description   { get; set; } = string.Empty;
	public string                   IssueDate     { get; set; }
	public string?                  DueDate       { get; set; }
	public Guid?                    CompanyId     { get; set; }

	public List<ProductDetail> Products     { get; set; }
	public List<CashProceed>  CashProceeds { get; set; }

	public decimal          DepositAmount    { get; set; }
	public decimal          WithdrawalAmount { get; set; }
	public decimal          TotalAmount      { get; set; }

	public CurrencyTypeEnum  CurrencyType { get; set; } = CurrencyTypeEnum.TL;
	public PaymentEnum       Payment      { get; set; } = PaymentEnum.InCash;
	public OperationTypeEnum Operation    { get; set; } = OperationTypeEnum.Sales;
	public StatusEnum        Status       { get; set; } = StatusEnum.Sent;

	public Guid  CustomerId      { get; set; }
	[JsonIgnore]
	public Customer Customer   { get;         set; }

}